staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -139,3 +151,74 @@ func TestDecode(t *testing.T) {
 		}
 	}
 }
+
+func TestDecodeFieldNameMatching(t *testing.T) {
+	type s struct {
+		LowerA int `cbor:"a"`
+		UpperB int `cbor:"B"`
+	}
+
+	for _, tc := range []struct {
+		name    string
+		decMode cbor.DecMode
+		in      []byte
+		want    s
+	}{
+		{
+			name:    "case-insensitive match",
+			decMode: modes.DecodeLax,
+			in:      hexDecode("A1614101"), // {"A": 1}
+			want: s{
+				LowerA: 1,
+			},
+		},
+		{
+			name:    "case-sensitive match",
+			decMode: modes.DecodeLaxMatchingCaseSensitive,
+			in:      hexDecode("A2614101614202"), // {"A": 1, "B": 2}
```

Comment:
Might as well use this input for the previous case (one case per mode): `cbor.UnknownFieldError` using `Decode` and nil error with `{LowerA: 0, UpperB: 2}` using `DecodeLax`.